### DISCUSSION TITLE
Assistant documentation not consistent with experience

### DISCUSSION THREAD

**Discussion by goseinthemachine** [zed-industries/zed#6943] [Feature Requests]
The docs sequence of events, posted below, did not happen when I first installed Zed and opened the assistant pane. It opened directly to the chat with assistant window and allowed me to type things in. Was confused for a bit then I discovered the `assistant: reset key` command. Since the docs are inconsistent with the experience should this be written as an issue? There are issues stating that the assistant isn't working but not about the onboarding of the setting up the assistant on initial interaction.

> Setup
> 
> When you first open the panel, Zed will ask you for an [OpenAI API key](https://platform.openai.com/account/api-keys), which is saved to your keychain.
> 
> 
> ![](https://zed.dev/img/assistant/openai-key.png)
> 
>

**Comment by jjcxdev** [comment]
I've run into the same issue, and I cannot figure out how to configure my api key at all.
Assistant is virtually useless for me right now.

**Reply by goseinthemachine** [reply]
Apparently from the other issues I've seen it is required to have credits in your OpenAI account for it to work properly.

**Reply by mrnugget** [reply]
Yes, that's correct. Sorry about that, we need to fix the docs.

Short version:

1. Create OpenAI Platform account
2. Add credits to the account
3. Create an API key
4. Open the assistant panel (it has to be focused)
5. Use command palette (`cmd-shift-p`) to run `assistant: reset key`
6. Enter the API key

**Reply by jjcxdev** [reply]
Yeah my issue isn't with my OpenAI API Key; I'm not using a free account. The issue stemmed from the fact that:

1. I wasn't prompted to enter my key as per the docs
2. The docs don't make it obvious that there are different command palette functionalities depending on which pane you are focused to
3. There is one small throw away line in the docs about resetting your key - this should be highlighted better and emphasized that if you aren't greeted with the api prompt, then you should run the reset command.

**Reply by mrnugget** [reply]
I updated the docs: https://zed.dev/docs/assistant-panel

**Reply by mrnugget** [reply]
I also just merged a PR that should fix the root problem: https://github.com/zed-industries/zed/pull/7197

Once this is released, I can revert my changes to the docs.

**Reply by jjcxdev** [reply]
That's significantly improved, there shouldn't be any confusion after that. Much thanks.

**Reply by mrnugget** [reply]
Thanks! Closing the discussion then. Feel free to reopen.
